gensim/sklearn_api/w2vmodel.py

```diff
@@ -83,4 +83,5 @@ def transform(self, words):
         return np.reshape(np.array(X), (len(words), self.size))
 
     def partial_fit(self, X):
-        raise NotImplementedError("'partial_fit' has not been implemented for W2VTransformer")
+        raise NotImplementedError("'partial_fit' has not been implemented for W2VTransformer since 'update()' function for Word2Vec class is experimental. "
```

Comment:
@chinmayapancholi13 thanks for the clarification.

So the concern is about initializing the vocabulary. That's a separate concern to incremental training, which is fully supported and not experimental (in fact, it's the same code as non-incremental training).

I don't think `partial_fit` is a high priority, but at the same time, users ask for it all the time, especially in combination with incremental vocab updates (which are indeed experimental). 

So I see two directions (both non-critical, nice-to-have) here:

1. support incremental training with a fixed vocabulary
  * implies finding a natural way to initialize vocabulary before hand
  * implies finding a natural way to control the learning rate (alpha) during the incremental calls.
2. support fully online training, incl. updating the vocabulary incrementally
  * implies changing the w2v algo(s) to support this (hashing trick with fixed hash space?)
  * still implies finding a way to control alpha